### PR TITLE
Feature: Added menu option to compress items into archive

### LINKED ISSUE #9947: Feature: Menu option to add selected items to archive
**Requirements**
- Add an option to the right click context menu to add the selected item to an archive
  - The menu option should say "Add to {_FolderName_}.zip"
- When multiple items are selected, there should be a different menu option that says "Add to archive"
  - This action should display a modal where the user can input the archive name. There should also be a cancel button.

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #9947 

**Validation**
How did you test these changes?
- [x] Built and ran the app

**Screenshots (optional)**

Screenshot (43)
Screenshot (42)
Screenshot (44)
Screenshot (46)

### CODE REVIEW

File: src/Files.App/Helpers/ContextFlyoutItemHelper.cs [hunk 1]
```diff
@@ -944,6 +945,22 @@ public static List<ContextMenuFlyoutItemViewModel> GetBaseItemMenuItems(BaseLayo
                     ShowInZipPage = true,
                 },
                 new ContextMenuFlyoutItemViewModel()
+                {
```
Comment (yaichenbaum) [inline src/Files.App/Helpers/ContextFlyoutItemHelper.cs:948]: It might be better if it was lower down in the menu

**Comment by yaichenbaum** [comment]
Looking good!

File: src/Files.App/Helpers/ContextFlyoutItemHelper.cs (hunk 1 above)
Comment (ferrariofilippo) [reply src/Files.App/Helpers/ContextFlyoutItemHelper.cs:948]: Below `Tag Section` is ok, or would it be better if I move it in the `More` section?

Comment (yaichenbaum) [reply src/Files.App/Helpers/ContextFlyoutItemHelper.cs:948]: Under the tag section is better than `more`.

Comment (ferrariofilippo) [reply src/Files.App/Helpers/ContextFlyoutItemHelper.cs:948]: Menu

Comment (yaichenbaum) [reply src/Files.App/Helpers/ContextFlyoutItemHelper.cs:948]: 👍

File: src/Files.App/Helpers/ContextFlyoutItemHelper.cs [hunk 2]
```diff
@@ -1030,14 +1031,30 @@ public static List<ContextMenuFlyoutItemViewModel> GetBaseItemMenuItems(BaseLayo
                     ShowInFtpPage = true,
                     SingleItemOnly = true,
                 },
-                new ContextMenuFlyoutItemViewModel()
+				new ContextMenuFlyoutItemViewModel()
```
Comment (yaichenbaum) [inline src/Files.App/Helpers/ContextFlyoutItemHelper.cs:1034]: Is spacing correct?

File: src/Files.Shared/Enums/FileOperationType.cs [hunk 3]
```diff
@@ -56,5 +56,10 @@ public enum FileOperationType : byte
         /// An item is being preparend for copy/move/drag
         /// </summary>
         Prepare = 10,
+
+        /// <summary>
+        /// An item has been compressed
```
Comment (yaichenbaum) [inline src/Files.Shared/Enums/FileOperationType.cs:61]:
```suggestion
        /// An item has been added to an archive
```

File: src/Files.App/Helpers/ZipHelpers.cs [hunk 4]
```diff
@@ -6,125 +6,182 @@
...
-                    entriesFinished++;
-                    float percentage = (float)((float)entriesFinished / (float)entriesAmount) * 100.0f;
-                    progressDelegate?.Report(percentage);
-                }
-            }
-        }
-    }
+	public static class ZipHelpers
+	{
+		public static async Task<bool> CompressSingleToArchive(string sourceFolder, string archive)
```
Comment (yaichenbaum) [inline src/Files.App/Helpers/ZipHelpers.cs:15]: Is it possible to have a single task that handles this logic? Ideally we can pass in sourceFolder or sourceFolders to the same task.

File: src/Files.App/Strings/en-US/Resources.resw [hunk 5]
```diff
@@ -2825,6 +2825,24 @@ We use App Center to track which settings are being used, find bugs, and fix cra
   <data name="BehaviorForOpeningFilesAndFolders" xml:space="preserve">
     <value>Behavior for opening files and folders</value>
   </data>
+  <data name="AddSingleItemToArchive.Text" xml:space="preserve">
+    <value>Add to {0}.zip</value>
+  </data>
+  <data name="AddToArchive.Text" xml:space="preserve">
```
Comment (yaichenbaum) [inline src/Files.App/Strings/en-US/Resources.resw:2831]:
```suggestion
  <data name="AddToArchive" xml:space="preserve">
```

File: src/Files.App/Strings/en-US/Resources.resw [hunk 6]
```diff
@@ -2825,6 +2825,24 @@ We use App Center to track which settings are being used, find bugs, and fix cra
   <data name="BehaviorForOpeningFilesAndFolders" xml:space="preserve">
     <value>Behavior for opening files and folders</value>
   </data>
+  <data name="AddSingleItemToArchive.Text" xml:space="preserve">
```
Comment (yaichenbaum) [inline src/Files.App/Strings/en-US/Resources.resw:2828]:
```suggestion
  <data name="AddSingleItemToArchive" xml:space="preserve">
```

File: src/Files.App/Strings/en-US/Resources.resw [hunk 7]
```diff
@@ -2825,6 +2825,24 @@ We use App Center to track which settings are being used, find bugs, and fix cra
...
+    <value>Add to {0}.zip</value>
+  </data>
+  <data name="AddToArchive.Text" xml:space="preserve">
+    <value>Add to archive</value>
+  </data>
+  <data name="CompressionCompletedText" xml:space="preserve">
```
Comment (yaichenbaum) [inline src/Files.App/Strings/en-US/Resources.resw:2834]:
```suggestion
  <data name="CompressionCompleted" xml:space="preserve">
```

File: src/Files.App/Strings/en-US/Resources.resw [hunk 8]
```diff
@@ -2825,6 +2825,24 @@ We use App Center to track which settings are being used, find bugs, and fix cra
...
+  </data>
+  <data name="CompressionSuccededText" xml:space="preserve">
+    <value>{0} has been compressed</value>
+  </data>
+  <data name="CompressionFailedText" xml:space="preserve">
+    <value>{0} couldn't be compressed</value>
+  </data>
+  <data name="CompressionInProgressText" xml:space="preserve">
+    <value>Compressing archive</value>
+  </data>
```
Comment (yaichenbaum) [inline src/Files.App/Strings/en-US/Resources.resw:2845]:
```suggestion
  <data name="CompressionSucceded" xml:space="preserve">
    <value>{0} has been compressed</value>
  </data>
  <data name="CompressionFailed" xml:space="preserve">
    <value>{0} couldn't be compressed</value>
  </data>
  <data name="CompressionInProgress" xml:space="preserve">
    <value>Compressing archive</value>
  </data>
```

File: src/Files.App/Strings/en-US/Resources.resw (hunk 6 above)
Comment (ferrariofilippo) [reply src/Files.App/Strings/en-US/Resources.resw:2828]: Should I change also `AddToArchive.Text` to `AddToArchive`?

Comment (yaichenbaum) [reply src/Files.App/Strings/en-US/Resources.resw:2828]: Yes 👍
Btw, is it posible to leave the conversations as unresloved so it's easier to find them later? Thank you 🙂

**Review by yaichenbaum — DISMISSED** [review]
LGTM

**Pull request merged**